yaksh/models.py

```diff
@@ -1129,6 +1154,113 @@ def has_lessons(self):
                 break
         return status
 
+    @classmethod
+    def _create_object_from_dict(cls, model_obj, model_type):
+        model_class = get_model_class(model_type)
+        new_obj, obj_create_status = model_class.objects.get_or_create(**model_obj)
+        return new_obj, obj_create_status
+
+    @classmethod
+    def load_yaml(cls, filepath, user):
+        file_list = {}
+        with zipfile.ZipFile(filepath,'r') as zfile:
+            for name in zfile.namelist():
+                file_content = zfile.read(name)
+                yaml_content = ruamel.yaml.safe_load(file_content)
+                if yaml_content.get('type') and yaml_content.get('type') == 'course':
+                    course = yaml_content
+                file_list.update(
+                    {name: yaml_content}
+                )
+        subitems = course.pop('subitems')
+        model_type = course.pop('type')
+        course.update({'creator': user})
+        new_course, course_create_status = cls._create_object_from_dict(course, 'course')
+        new_mod_list = []
```

Comment:
name can be simply modules or new_modules or new_mods. No need of _list here.
There are few other places with similar thing.

Reply:
Done